### PR TITLE
feat: Replace Web Notification rendering engine from gtmpl to Vue - EXO-66404

### PR DESCRIPTION
**Pull request by rdenarie**
This change will allow to not having a gtmpl template associated for some Web Notifications to use Frontend side display instead. This will ensure a better UX, evolutivity and maintainability.

### CODE REVIEW

File: agenda-services/src/main/java/org/exoplatform/agenda/util/NotificationUtils.java [hunk 1]
```diff
@@ -179,6 +179,8 @@ public class NotificationUtils {
 
   public static final String                                 STORED_PARAMETER_EVENT_STATUS                  = "eventStatus";
 
+  private static final String                                 STORED_PARAMETER_EVENT_IS_CREATOR = "isCreator";
```
Comment (ahamdi) [inline agenda-services/src/main/java/org/exoplatform/agenda/util/NotificationUtils.java]: Could you format this line ?

File: agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/AgendaEventPlugin.vue [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    spaceName() {
+      return this.space?.prettyName;
+    },
+    message() {
+      const eventTitle = this.notification?.parameters?.eventTitle;
+      let message = '';
+      const agendaName = this.spaceName;
+
+      if (this.notification?.parameters?.EVENT_MODIFICATION_TYPE === 'ADDED') {
+        message = this.$t('Notification.agenda.event.created', {0: `<span class="font-weight-bold">${eventTitle}</span>`, 1: `<a class="space-name font-weight-bold">${agendaName}</a>`});
```
Comment (ahamdi) [inline agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/AgendaEventPlugin.vue]: We may have a dedicated function for the block inside those if/else as it is repeated, WDYT ?

File: agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/EventReminderPlugin.vue [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    space: null,
+  }),
+  props: {
+    notification: {
+      type: Object,
+      default: null,
+    },
+  },
+  created() {
+    console.log(this.notification);
```
Comment (ahamdi) [inline agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/EventReminderPlugin.vue]: Could you remove this logging statement ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: agenda-services/src/main/java/org/exoplatform/agenda/util/NotificationUtils.java (hunk 1 above)
Comment (rdenarie) [reply agenda-services/src/main/java/org/exoplatform/agenda/util/NotificationUtils.java]: done

File: agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/AgendaEventPlugin.vue (hunk 2 above)
Comment (rdenarie) [reply agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/AgendaEventPlugin.vue]: I am unable to completly remove the if else.  I reduce the code duplication by using a dedicated function

File: agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/EventReminderPlugin.vue (hunk 3 above)
Comment (rdenarie) [reply agenda-webapps/src/main/webapp/vue-app/agenda-notifications/components/EventReminderPlugin.vue]: done

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**